### PR TITLE
Add MeasureEx

### PR DESCRIPTION
**Pull request by ppease**
### What does this Pull Request accomplish?

Add the method `niSync_MeasureFrequencyEx` for the measure test panel for NI-Sync.

### Why should this Pull Request be merged?

Adding methods need for the measure test panel for NI-Sync.

### What testing has been done?

[ RUN      ] NiSyncDriverApiTest.MeasureFrequencyExOnTerminalWithNoFrequency_ReturnsNoFrequency
[       OK ] NiSyncDriverApiTest.MeasureFrequencyExOnTerminalWithNoFrequency_ReturnsNoFrequency (106 ms)
[ RUN      ] NiSyncDriverApiTest.MeasureFrequencyExOnOscillatorWithFrequency_ReturnsFrequency
[       OK ] NiSyncDriverApiTest.MeasureFrequencyExOnOscillatorWithFrequency_ReturnsFrequency (105 ms)
[----------] 18 tests from NiSyncDriverApiTest (750 ms total)

### CODE REVIEW

File: source/tests/system/nisync_driver_api_tests.cpp [hunk 1]
```diff
@@ -540,6 +557,36 @@ TEST_F(NiSyncDriverApiTest, AttributeSet_GetAttributeViString_ReturnsValue)
...
+  ViStatus viStatus;
+  ViConstString srcTerminal = NISYNC_VAL_OSCILLATOR;
+  ViReal64 duration = 0.1; // duration is in seconds
+  ViUInt32 decimationCount = 0; // ignored for 6674
+  ViReal64 actualDuration, frequency, frequencyError;
+  auto grpcStatus = call_MeasureFrequencyEx(srcTerminal, duration, decimationCount, &actualDuration, &frequency, &frequencyError, &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+  EXPECT_GT(actualDuration, 0);
```
Comment (ppease) [inline source/tests/system/nisync_driver_api_tests.cpp:586]: For both of these values I'm just checking that they are greater than zero.  In reality it should be really close to 0.1 for the duration and really close to 10 for the frequency.

File: source/tests/system/nisync_driver_api_tests.cpp [hunk 2]
```diff
@@ -249,6 +249,23 @@ class NiSyncDriverApiTest : public ::testing::Test {
     return grpcStatus;
   }
 
+  ::grpc::Status call_MeasureFrequencyEx(ViConstString srcTerminal, ViReal64 duration, ViUInt32 decimationCount, ViReal64* actualDurationOut, ViReal64* frequencyOut, ViReal64* frequencyErrorOut, ViStatus* viStatusOut)
```
Comment (yixinguo2) [inline source/tests/system/nisync_driver_api_tests.cpp:252]: This line is a bit long, put parameters on separate lines?

**Review by yixinguo2 — APPROVED** [review]

File: source/tests/system/nisync_driver_api_tests.cpp (hunk 2 above)
Comment (ppease) [reply source/tests/system/nisync_driver_api_tests.cpp]: Fixed in commit 5 and 6

File: generated/nisync/nisync.proto [hunk 3]
```diff
@@ -25,6 +25,7 @@ service NiSync {
   rpc DisconnectSWTrigFromTerminal(DisconnectSWTrigFromTerminalRequest) returns (DisconnectSWTrigFromTerminalResponse);
```
Comment (ppease) [inline generated/nisync/nisync.proto:25]: I'm guessing my changes will conflict with Zach's recent PR to add some additional get/set methods.  I'll wait for his to go in and resolve conflicts.

**Review by tkrebes — APPROVED** [review]

File: source/tests/system/nisync_driver_api_tests.cpp (hunk 1 above)
Comment (reckenro) [reply source/tests/system/nisync_driver_api_tests.cpp:607]: This seems fine to me. More of an FYI, but Google Test has some assertions for floats / doubles that assert the values are "close" to each other and one that takes in an error bound (see https://github.com/google/googletest/blob/master/docs/advanced.md#floating-point-comparison).

I expect just checking greater than zero is fine here but those are some alternative assertions you can use.

**Review by reckenro — APPROVED** [review]

File: source/tests/system/nisync_driver_api_tests.cpp (hunk 1 above)
Comment (jonlugoatx) [reply source/tests/system/nisync_driver_api_tests.cpp:607]: For cases where you would like to test specific values or corner cases, you may want to consider adding an integration test that uses a mocked version of your service. Also the benefit in doing so is that the integration tests run on the CI while the system tests do not. This test looks good as is though.

**Review by jonlugoatx — APPROVED** [review]

File: source/tests/system/nisync_driver_api_tests.cpp (hunk 1 above)
Comment (ppease) [reply source/tests/system/nisync_driver_api_tests.cpp:699]: I really just wanted to do sanity testing here to make sure things look reasonable.  Thanks for the suggestions about floating point and the integration tests.  I'll keep that in mind for future changes.  I'm going to leave this as is for now.
